client/driver/qemu.go

```diff
@@ -381,9 +431,26 @@ func (h *qemuHandle) Signal(s os.Signal) error {
 	return fmt.Errorf("Qemu driver can't send signals")
 }
 
-// TODO: allow a 'shutdown_command' that can be executed over a ssh connection
-// to the VM
 func (h *qemuHandle) Kill() error {
+	// If a graceful shutdown was requested at task start time,
+	// monitorPath will not be empty
+	if h.monitorPath != "" {
+		monitorSocket, err := net.Dial("unix", h.monitorPath)
+		if err == nil {
+			defer monitorSocket.Close()
+			h.logger.Printf("[DEBUG] driver.qemu - sending graceful shutdown command to qemu monitor socket at %s", h.monitorPath)
+			_, err = monitorSocket.Write([]byte(qemuGracefulShutdownMsg))
+			if err == nil {
+				return nil
+			}
+			h.logger.Printf("[WARN] driver.qemu - failed to send '%s' to monitor socket '%s': %s", qemuGracefulShutdownMsg, h.monitorPath, err)
```

Comment:
Using `%q` instead of `'%s'` will properly escape and quote strings for what it's worth. Not a big deal to change as we're not consistent either, but `%q` is pretty handy.